diff --git a/usr/src/lib/crt/Makefile.com b/usr/src/lib/crt/Makefile.com
index 8384fc09fa..12dde06185 100644
--- a/usr/src/lib/crt/Makefile.com
+++ b/usr/src/lib/crt/Makefile.com
@@ -60,19 +60,8 @@ $(COMMON_CRT) $(VALUES) := CFLAGS64 += $(C_PICFLAGS64)
 # intentional double exit
 SMOFF += unreachable
 
-COMPATLINKS=	usr/ccs/lib/values-Xa.o \
-		usr/ccs/lib/values-Xc.o \
-		usr/ccs/lib/values-Xs.o \
-		usr/ccs/lib/values-Xt.o \
-		usr/ccs/lib/values-xpg4.o \
-		usr/ccs/lib/values-xpg6.o
-
-COMPATLINKS64=	usr/ccs/lib/$(MACH64)/values-Xa.o \
-		usr/ccs/lib/$(MACH64)/values-Xc.o \
-		usr/ccs/lib/$(MACH64)/values-Xs.o \
-		usr/ccs/lib/$(MACH64)/values-Xt.o \
-		usr/ccs/lib/$(MACH64)/values-xpg4.o \
-		usr/ccs/lib/$(MACH64)/values-xpg6.o
+COMPATLINKS=
+COMPATLINKS64=
 
 $(ROOT)/usr/ccs/lib/values-Xa.o:=	COMPATLINKTARGET=../../lib/values-Xa.o
 $(ROOT)/usr/ccs/lib/values-Xc.o:=	COMPATLINKTARGET=../../lib/values-Xc.o
