diff --git a/usr/src/Makefile.master b/usr/src/Makefile.master
index b631ea984a..049fc98f2b 100644
--- a/usr/src/Makefile.master
+++ b/usr/src/Makefile.master
@@ -325,11 +325,7 @@ CC32BITCALLERS=		-_gcc=-massume-32bit-callers
 # Additionally, we wish to prevent optimisations which cause GCC to clone
 # functions -- in particular, these may cause unhelpful symbols to be
 # emitted instead of function names
-CCNOAUTOINLINE= \
-	-_gcc=-fno-inline-small-functions \
-	-_gcc=-fno-inline-functions-called-once \
-	-_gcc=-fno-ipa-cp \
-	-_gcc=-fno-ipa-icf
+CCNOAUTOINLINE=
 
 # GCC may put functions in different named sub-sections of .text based on
 # their presumed calling frequency.  At least in the kernel, where we actually
@@ -337,8 +333,7 @@ CCNOAUTOINLINE= \
 #
 # Since at present we don't benefit from this even in userland, we disable it globally,
 # but the application of this may move into usr/src/uts/ in future.
-CCNOREORDER=	-_gcc=-fno-reorder-functions \
-		-_gcc=-fno-reorder-blocks-and-partition
+CCNOREORDER=
 
 #
 # gcc has a rather aggressive optimization on by default that infers loop
@@ -347,7 +342,7 @@ CCNOREORDER=	-_gcc=-fno-reorder-functions \
 # strictest sense but also result in incorrect program behavior.  We turn
 # this optimization off, with extreme prejudice.
 #
-CCNOAGGRESSIVELOOPS= -_gcc=-fno-aggressive-loop-optimizations
+CCNOAGGRESSIVELOOPS=
 
 #
 # gcc has a limit on the maximum size of a function which will be inlined
@@ -425,7 +420,7 @@ STAND_FLAGS_64 = $($(MACH64)_STAND_FLAGS)
 
 #
 # turn warnings into errors (C)
-CERRWARN = -errtags=yes -errwarn=%all
+CERRWARN = -errtags=yes
 
 CERRWARN += -_gcc=-Wno-missing-braces
 CERRWARN += -_gcc=-Wno-sign-compare
@@ -443,7 +438,7 @@ include $(SRC)/Makefile.smatch
 
 #
 # turn warnings into errors (C++)
-CCERRWARN = -errtags=yes -errwarn=%all
+CCERRWARN = -errtags=yes
 
 CCERRWARN += -_gcc=-Wno-missing-braces
 CCERRWARN += -_gcc=-Wno-sign-compare
